modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/hdb/hdbUtils.js

```diff
@@ -18,7 +18,7 @@ exports.getResultSetValueByDataTypeAndRowNumber = function (resultSet, dataType,
         case "INTEGER":
             return resultSet.getInt(colNumber);
         case "BIGINT":
-            return resultSet.getLong(colNumber);
+            return new Int64(resultSet.getLong(colNumber));
```

Comment:
I think you would have to pass the `resultSet` variable into the Java side of `Int64` and get a `long` from there. With how big numbers did you test it? This would potentially have problems with bigger numbers coming from the DB as the GraalJS would convert them to JS numbers